Superbridge docs (#197)
hyperlane-xyz/v3-docs: docs/guides/deploy-warp-route-UI.mdx

# How to Deploy a Bridge UI for Your Hyperlane Warp Route

> After you've successfully [deployed a Warp Route](/docs/guides/deploy-warp-route.mdx), you may want to deploy a front-end UI to faciliate interchain token transfers. The easiest ways to get going with this are to,

Reviewer: ```suggestion
After you've successfully [deployed a Warp Route](/docs/guides/deploy-warp-route.mdx), you may want to deploy a front-end UI to facilitate interchain token transfers. There are two popular options:
```

Besides warning users in the UI about ensuring they have gas, you can also faucet them some native gas tokens to improve their initial experience. To achieve this faucet, you can modify the Warp Route contracts to hold some balance of the native token and share that with recipients. In addition, you could build an off-chain service which just observes for [`ReceivedTransferRemote events`](https://github.com/hyperlane-xyz/hyperlane-monorepo/blob/main/solidity/contracts/token/libs/TokenRouter.sol) and transfers the native gas token to the recipient.

:::tip

<TypescriptSdkPlaceholder />

:::

## Superbridge

> [Superbridge](https://superbridge.app) is a custom bridge provider, they handle hosting bridge frontends for hundreds of rollups, various tokens and different ecosystems. Getting a Superbridge for your warp route allows for a customized & hassle free approach to operating your bridge. They've also built a handy feature that allows for self-service testing of recently deployed warp routes.

Reviewer: 'approach to operating your bridge' make me think it's more than a front-end. Is there a more precise way to describe this?

Author: Yeah `operating` is doing a lot of heavy lifting here. What I mean is,

- hosting
- managing token lists
- user support 
- bug fixes
- managed upgrades (ie new features)

Not sure we need to outline all there in the Hyperlane docs we probably need a dedicated superbridge.app/hyperlane page that outlines everything offered.

### Testing your warp route

1. Navigate to the [Superbridge Hyperlane Playground](https://hyperlane.superbridge.app)
2. Click the settings cog and then click Customize ![](/img/deploy-warp-ui/superbridge-settings.png)
3. Paste in the YAML file you generated when deploying your warp routes ![](/img/deploy-warp-ui/superbridge-paste-yaml.png)
4. The specified tokens and networks will now be available for bridging.

> Note, if any tokens or networks are missing after you've pasted in the YAML file, it's possible Superbridge doesn't support this network just yet. [Reach out](mailto:support@superbridge.app) to them to see what the timeline is for adding support.

Reviewer 2: nit
```suggestion
:::note
If any tokens or networks are missing after you've pasted in the YAML file, it's possible Superbridge doesn't support this network just yet. [Reach out](mailto:support@superbridge.app) to them to see what the timeline is for adding support.
:::
```

### Production

> To bring your warp route to production with Superbridge, [reach out to the team](mailto:alex@superbridge.app)! They're super responsive and get can generally get you up and running in under 24 hours.

Reviewer: ```suggestion
To bring your warp route to production with Superbridge, [reach out to the team](mailto:alex@superbridge.app)! They can typically get a front-end up and running in about a day.
```

Reviewer: I appreciate the enthusiasm @AlexBHarley but for your own sanity, maybe don't promise people instant results or they'll start to expect it 😝

> A couple of their most popular Hyperlane powered bridge frontends include [Renzo](https://renzo.superbridge.app) and [Elixir](https://elixir.superbridge.app), if you're interested in checking out what the end product could look like.

Reviewer: Let's be consistent in the docs with 'front-end' vs 'frontend'. I prefer the former and I think that's what we've been using
